_posts/2019-05-10-continuous-deployment-the-first-step-on-the-road-to-recovery.markdown

```diff
@@ -0,0 +1,63 @@
+
+Your branching strategy is a fundamental decision for development teams and has profound effects upon your working practices.  Making a poor choice has knock on effects that can limit improvements in other areas.  
+
+The goal of [continuous delivery](https://continuousdelivery.com/) is:
+> To make deployments—whether of a large-scale distributed system, a complex production environment, an embedded system, or an app—predictable, routine affairs that can be performed on demand.
+> We achieve all this by ensuring our code is always in a deployable state
+
+Or more simply: “if it hurts, do it more often, and bring the pain forward.”
+
+# Damming the Flow
+
+A common branching strategy adopted by many teams is git flow.  
+
+[!GitFlow diagram](../img/continuous-deployment-the-first-step)*
+
+If you aren't familiar with git flow, [Atlassian have an excellent introduction](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).
+
+The goals for GitFlow are in direct conflict with the goals of Continuous Delivery and maximising flow, as [highlighted by the originator](https://nvie.com/posts/a-successful-git-branching-model/):
+> We consider origin/master to be the main branch where the source code of HEAD always reflects a production-ready state.
+> We consider origin/develop to be the main branch where the source code of HEAD always reflects a state with the latest delivered development changes for the next release.
+
+Maintaining multiple long-lived branches are an inhitibor to flow.  Changes queue on a branch, waiting for release to production.  They add complexity in understanding the divergence between branches, increasing the risk of regressions and merge hell.  This risk grows exponentially with the number of branches maintained.  Furthermore, the increased complexity of managing multiple branches, and the batching of commits, increases the complexity of a release and therefore the associated risk.  As a result, this leads to a downward spiral of increased batching, complexity and risk.
+
+Teams choose git flow for well intentioned reasons.  Faith in the codebase is low, caused by insufficient feedback on changes provided by a comprehensive automated test suite.  This leads to regular issues being found, further reducing faith.  Teams want control over feature releases: when they are confident of the quality. Batching and branching are an attempt to provide the stability to regain control.  However, it [has been shown](https://continuousdelivery.com/evidence-case-studies/) that these practices inhibit, rather than enable, flow.
+
+# Releasing the pressure
+
+A good practice for for [The First Way](https://itrevolution.com/the-three-ways-principles-underpinning-devops/) is [one piece flow](https://www.kaizenworld.com/kaizen/one-piece-flow.html).  Work on single increments of value, delivering them as soon as possible.  
```

Comment:
You probably need to introduce the Three Ways first.

'X said the Three Ways underpin DevOps, the first of which is Y. Here I explain why Y is important. A principle Z (from something something) is a method of reinforcing Y. blah blah.

Reply:
Again, assuming knowledge here - the referenced link gives details on the three ways if required.

Reply:
I think problem there is you already have a lot of links to other concepts. That's a lot of effort to just to be able to parse this post.

Who do you see the article being for? because it's hard to gauge at the moment.

Reply:
>You probably need to introduce the Three Ways first.

I agree with this because my [re-wording suggestion](https://github.com/MechanicalRock/mechanicalrock.github.io/pull/39#discussion_r286417884) indicates I was obviously unaware of it.

Reply:
Introduced in the first paragraph.  With a reference.

People need to read further first if required.